src/libertem/udf/crystallinity.py

```diff
@@ -0,0 +1,84 @@
+import functools
+
+import numpy as np
+
+from libertem.common.buffers import BufferWrapper
+from libertem.masks import _make_circular_mask
+
+
+def make_result_buffers():
+    return {
+        'intensity': BufferWrapper(
+            kind="nav", dtype="float32"
+        ),
+    }
+
+
+def init_fft(partition, rad_in, rad_out, real_center, real_rad):
+    real_mask = 1-1*_make_circular_mask(
+        real_center[1], real_center[0], partition.shape.sig[1],
+        partition.shape.sig[0], real_rad
+    )
+    fourier_mask_out = 1*_make_circular_mask(
+        partition.shape.sig[1]*0.5, partition.shape.sig[0]*0.5,
+        partition.shape.sig[1], partition.shape.sig[0], rad_out
+    )
+    fourier_mask_in = 1*_make_circular_mask(
+        partition.shape.sig[1]*0.5, partition.shape.sig[0]*0.5,
+        partition.shape.sig[1], partition.shape.sig[0], rad_in
+    )
+    fourier_mask = fourier_mask_out - fourier_mask_in
+
+    kwargs = {
+        'real_mask': real_mask,
+        'fourier_mask': fourier_mask,
+    }
+    return kwargs
+
+
+def fft(frame, real_mask, fourier_mask, intensity):
+    intensity[:] = np.sum(np.fft.fftshift(abs(np.fft.fft2(frame*real_mask)))*fourier_mask)
+    return
+
+
+def run_analysis_crystall(ctx, dataset, rad_in, rad_out, real_center=None, real_rad=None):
+    """
+    Return a value after integration of Fourier spectrum for each frame over ring.
+    Parameters
+    ----------
+    ctx: Context
+        Context class that contains methods for loading datasets,
+        creating jobs on them and running them
+
+    dataset: DataSet
+        A dataset with 1- or 2-D scan dimensions and 2-D frame dimensions
+    rad_in: int
+        Inner radius of a ring mask for the integration in Fourier space
+
+    rad_out: int
+        Outer radius of a ring mask for the integration in Fourier space
+
+    real_center: tuple
+        (x,y) - coordinates of a center of a circle for a masking out zero-order peak in real space
+
+    real_rad: int
+        (x,y) - radius of circle for a masking out zero-order peak in real space
```

Comment:
I think `(x,y)` for `real_rad` is not quite right...

Also, is `(x,y)` correct for `real_center`? In the `init_fft` function above, `_make_circular_mask` is called with `real_center[1], real_center[0]`. Compare this with the [signature of `_make_circular_mask`](https://github.com/LiberTEM/LiberTEM/blob/a3f54b1e531a248d74f0fe381b6df0068531c398/src/libertem/masks.py#L6) which has the parameters ordered as `centerX, centerY` (for historical reasons, really...).